go.mod

```diff
@@ -15,19 +15,26 @@ require (
 	github.com/google/uuid v1.1.1
 	github.com/gorilla/websocket v1.4.2
 	github.com/iotexproject/iotex-proto v0.4.3
+	github.com/irisnet/service-sdk-go v0.0.0-20201030091855-7f57f83f8c6c
 	github.com/jinzhu/gorm v1.9.12
 	github.com/magiconair/properties v1.8.1
 	github.com/ontio/ontology-go-sdk v1.11.1
 	github.com/pierrec/xxHash v0.1.5 // indirect
 	github.com/pkg/errors v0.9.1
 	github.com/smartcontractkit/chainlink v0.9.5
-	github.com/spf13/cobra v0.0.5
+	github.com/spf13/cobra v1.0.0
 	github.com/spf13/viper v1.7.1
 	github.com/status-im/keycard-go v0.0.0-20190424133014-d95853db0f48
 	github.com/stretchr/testify v1.6.1
+	github.com/tendermint/tendermint v0.33.4
 	github.com/tidwall/gjson v1.6.1
 	go.uber.org/zap v1.16.0
-	google.golang.org/grpc v1.27.0
+	google.golang.org/grpc v1.31.1
 	gopkg.in/gormigrate.v1 v1.6.0
 	launchpad.net/gocheck v0.0.0-20140225173054-000000000087 // indirect
 )
+
+replace (
+	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.2-alpha.regen.4
+	github.com/tendermint/tendermint => github.com/bianjieai/tendermint v0.34.0-irita-200930
+)
```

Comment:
I'm a bit cautious about using replace on too many dependencies. What's the reason we need to replace them here? Can we switch the dependency directly instead of replacing them?

Reply:
The main reason is that we need to replace them for compatibility. For one, we forked and improved it; For another, we indirectly depend on it but there exist compatibility issues.

Reply:
I had a discussion with a couple of my colleagues and we came to the conclusion that can't use `replace` like this in a production system.

We have two options here:
a) Have these changes merged into the upstream dependency
b) Import the replacement dependency explicitly

Reply:
@boxhock  Thanks for the suggestion from your team. Here are my points:

 a) The upstream project does not need the improvement which is required for us due to some critical reasons
 b) The replacement dependency can not be imported directly because it uses the same module path as upstream.

BTW, on the Go official wiki https://github.com/golang/go/wiki/Modules#when-should-i-use-the-replace-directive, there is not any implication that the replace directive is associated with the production security. 

If this is still your concern, I will consider other possible solutions.